jquery.flot.js

```diff
@@ -2536,14 +2536,19 @@ Licensed under the MIT license.
                 var points = series[i].datapoints.points;
                 ps = series[i].datapoints.pointsize;
 
+                // if the data is coming from positive -> negative, reverse the comparison
+                const comparer = points[points.length - ps] < points[0]
+                    ? function (x1, x2) { return x1 > x2 }
+                    : function (x1, x2) { return x2 > x1 };
+
                 // do not interpolate outside the bounds of the data.
-                if (posX < points[0]) {
+                if (comparer(posX, points[0])) {
```

Comment:
What seems wrong with this line, to me, is that it's incomplete, not that we have to deal with data "going in a different direction". Shouldn't this simply change to:

if (posX < points[0] || posX > points[points.length - 1])

Reply:
The problem is that points[0] might not be the smallest value. Specifically when the axis is reversed, you still want the data to come in from left-to-right so it should be acceptable that data would come in from largest to smallest 
eg.
XValues = [ 3, 2, 1, 0, -1, -2, -3]
should be a valid array of xvalues.
In this case posX < points[0] will fail, as will poxX > points[points.length - 1]

Reply:
Ah, of course (duh).